Ch 444/ch 669 change endpoint to update user data

NatalliaBulyha opened the pull request.

Commit:
CH-444/CH-669 changed valid logic

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiZmMwNDg2MmU3MGZjNGJlZjk2MjQ1ZmYzZGI2ZDdlYmUiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 changed valid logic

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiZmMwNDg2MmU3MGZjNGJlZjk2MjQ1ZmYzZGI2ZDdlYmUiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 added AuthenticationException("Wrong old password entered.")

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiZmMwNDg2MmU3MGZjNGJlZjk2MjQ1ZmYzZGI2ZDdlYmUiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 changed isValid

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiZmMwNDg2MmU3MGZjNGJlZjk2MjQ1ZmYzZGI2ZDdlYmUiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 changed the password check in the method updateUserData

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiZmMwNDg2MmU3MGZjNGJlZjk2MjQ1ZmYzZGI2ZDdlYmUiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 changed the password check in the method updateUserData

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiZmMwNDg2MmU3MGZjNGJlZjk2MjQ1ZmYzZGI2ZDdlYmUiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 changed validation for firstName and lastName

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiZmMwNDg2MmU3MGZjNGJlZjk2MjQ1ZmYzZGI2ZDdlYmUiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 changed optional fields validation

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiYjU3ZWZkNzY5YTZlNDAxZGEzZjM2ZTg2Mzc4ZTA1MzEiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 changed check logic in service

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiYjU3ZWZkNzY5YTZlNDAxZGEzZjM2ZTg2Mzc4ZTA1MzEiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 added ValidFieldsValueMatch

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiYjU3ZWZkNzY5YTZlNDAxZGEzZjM2ZTg2Mzc4ZTA1MzEiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 added ValidFieldsValueMatch

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiYjU3ZWZkNzY5YTZlNDAxZGEzZjM2ZTg2Mzc4ZTA1MzEiLCJwIjoiaiJ9

Commit:
CH-444/CH-669 changed ValidFieldsValueMatch

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiYjU3ZWZkNzY5YTZlNDAxZGEzZjM2ZTg2Mzc4ZTA1MzEiLCJwIjoiaiJ9

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/dto/user/UserInformationRequest.java:
```diff
@@ -15,16 +16,24 @@
     private String email;
     private String oldPassword;
     private String newPassword;
+    private String passwordReset;
```
could we rename it, because it confuses. Maybe "RepeatedNewPassword"

NatalliaBulyha replied:
ok

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/dto/user/UserInformationRequest.java:
```diff
@@ -15,16 +16,24 @@
     private String oldPassword;
     private String newPassword;
+    private String passwordReset;
     @ValidOptionalFieldBirthday
```
could we rename it to "ValidOptionalBirthday"? or something else?

NatalliaBulyha replied:
ok

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/dto/user/UserInformationRequest.java:
```diff
@@ -15,16 +16,24 @@
-    @ValidFirstNameAndLastName
+    @ValidFirstNameAndLastName(message = "Invalid firstName. FirstName must be not null, greater than " +
+            "or equal to 2 and less than or equal to 32, letters only.")
     private String firstName;
```
propose to rename the class to "@ValidName" and left the default message only :"Invalid name. The name must not be null, must be greater than or equal to 2, and less than or equal to 32 characters, letters only."

NatalliaBulyha replied:
ok

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java:
```diff
@@ -139,31 +140,31 @@ public UserInformationResponse updateUserData(UserInformationRequest userInforma
 			if (userRepository.existsByEmail(userInformationDto.getEmail()) ) {
 				throw new DataAlreadyExistException(String.format("Email is already in use by another user. " +
 						"Forbidden to change mail to %s.", userInformationDto.getEmail()));
 			}
```
Propose to change exception message to :"Email  %s is already in use", userInformationDto.getEmail().  It's sufficient to understand the reason for the exception.

NatalliaBulyha replied:
ok

DariaBoo commented on src/main/java/com/cozyhome/onlineshop/validation/impl/OptionalFieldBirthdayConstraintValidator.java:
```diff
@@ -12,10 +12,13 @@ public void initialize(ValidOptionalFieldBirthday constraintAnnotation) {
+        }
+        if (birthday.isEmpty()) {
+            return true;
         }
```
this two if statements a little bit confused, in one case it's true, in another case false

NatalliaBulyha replied:
is an optional field, an empty string is allowed, but null is not allowed

Commit:
CH-444/CH-669 correction of comments

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiODM0MjFmYTZjNzAyNGNhYTk5ODFmMDg4Y2UwYWRlMzYiLCJwIjoiaiJ9

Commit:
Merge remote-tracking branch 'origin/main' into CH-444/CH-669

# Conflicts:
#	src/main/java/com/cozyhome/onlineshop/userservice/security/service/impl/UserServiceImpl.java
#	src/main/java/com/cozyhome/onlineshop/validation/impl/OptionalFieldBirthdayConstraintValidator.java

Commit:
CH-444/CH-669 resolve conflicts

jira [simplification of validation for changing the password in the personal account] https://cozyhometeam.atlassian.net/browse/CH-669?atlOrigin=eyJpIjoiODM0MjFmYTZjNzAyNGNhYTk5ODFmMDg4Y2UwYWRlMzYiLCJwIjoiaiJ9
